active/0003-named-template-blocks.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 2015/04/12
+- RFC PR: 
+- ember-cli Issue: 
+
+# Summary
+
+Introduce a new Handlebars syntax that would denote beginning and end of a template block. 
+
+# Motivation
+
+In Ember 2.0, we're moving towards components becoming primary mechanisms for encapsulating functionality with templates being the mechanism that wires these components together. It would be helpful to allow developers the ability to customize certain portion of component's template from it's block template without having to extend a component or overwrite it's layout.
+
+Currently, we have the ability to do this but it's limited to `{{else}}` helper.
+
+For example, we can customize what `{{each}}` helper presents when the array passed to it is empty.
+
+```
+{{#each model as |item|}}
+	{{item.name}}
+{{else}}
+  	No items are available.
+{{/each}}
+```
+
+`{{if}}` helper has the same behaviour.
+
+We are currently lacking the ability to specify a custom template block that we can consume inside of the component or a helper. This RFC proposes that we introduce the ability to specify custom named template blocks that can be used to customize the default behaviour of a component or a helper. 
+
+A table component is a good example of a component that could benefit from this API. Let's consider Addepar's [Ember Table Component](https://github.com/Addepar/ember-table). *Ember Table* has 3 distinct areas that a user might want to cusomize while using the component, namely header, body & footer. 
+
+Currently, *Ember Table*'s component layout looks like this
+
+```
+{{#if controller.hasHeader}}
+  {{view Ember.Table.HeaderTableContainer}}
+{{/if}}
+{{view Ember.Table.BodyTableContainer}}
+{{#if controller.hasFooter}}
+  {{view Ember.Table.FooterTableContainer}}
+{{/if}}
+{{view Ember.Table.ScrollContainer}}
+{{view Ember.Table.ColumnSortableIndicator}}
+```
+
+To customize any of these areas, the developer has to extend the component, change specify a new layout and make changes inside of this new layout. If we had the ability to specify custom named template blocks then we could allow the developer to customize one of these areas from the template where the component is being used. 
+
+Here is what that might look like,
+
+```
+{{#ember-table}}
```

Comment:
@stefanpenner this is a different use case and a different RFC, yielded helpers are about the block placing the content.   Named templates are about the parent controlling the rendering via its layout.  We need to keep these things separate.